src/datatype/verify.rs

```diff
@@ -45,47 +47,59 @@ impl FromStr for KeyType {
     }
 }
 
+
 #[derive(PartialEq, Eq, Debug, Clone)]
-pub enum SignatureType {
+pub enum SigType {
     Ed25519,
     RsaSsaPss,
 }
 
-impl serde::Deserialize for SignatureType {
-    fn deserialize<D: serde::Deserializer>(de: D) -> Result<Self, D::Error> {
-        if let json::Value::String(ref s) = serde::Deserialize::deserialize(de)? {
-            s.parse().map_err(|err| serde::de::Error::custom(format!("unknown SignatureType: {}", err)))
+impl Deserialize for SigType {
+    fn deserialize<D: Deserializer>(de: D) -> Result<Self, D::Error> {
+        if let json::Value::String(ref s) = Deserialize::deserialize(de)? {
+            s.parse().map_err(|err| SerdeError::custom(format!("unknown SigType: {}", err)))
         } else {
-            Err(serde::de::Error::custom("unknown SignatureType"))
+            Err(SerdeError::custom("unknown SigType"))
         }
     }
 }
 
-impl serde::Serialize for SignatureType {
-    fn serialize<S: serde::Serializer>(&self, ser: S) -> Result<S::Ok, S::Error> {
+impl Serialize for SigType {
+    fn serialize<S: Serializer>(&self, ser: S) -> Result<S::Ok, S::Error> {
         ser.serialize_str(match *self {
-            SignatureType::Ed25519   => "ed25519",
-            SignatureType::RsaSsaPss => "rsassa-pss"
+            SigType::Ed25519   => "ed25519",
+            SigType::RsaSsaPss => "rsassa-pss"
         })
     }
 }
 
-impl FromStr for SignatureType {
+impl FromStr for SigType {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s {
-            "ed25519"    => Ok(SignatureType::Ed25519),
-            "rsassa-pss" => Ok(SignatureType::RsaSsaPss),
-            _ => Err(Error::UptaneInvalidSignatureType(s.to_string()))
+            "ed25519"    => Ok(SigType::Ed25519),
+            "rsassa-pss" => Ok(SigType::RsaSsaPss),
+            _ => Err(Error::UptaneInvalidSigType(s.to_string()))
         }
     }
 }
 
-impl SignatureType {
+impl From<KeyType> for SigType {
+    fn from(keytype: KeyType) -> Self {
+        match keytype {
+            KeyType::Ed25519 => SigType::Ed25519,
+            KeyType::Rsa     => SigType::RsaSsaPss,
```

Comment:
If all we know is that it is `KeyType::Rsa`, how can we delegate to multiple SigTypes?